writeengine/wrapper/writeengine.cpp

```diff
@@ -837,7 +837,18 @@ int WriteEngineWrapper::deleteRow(const TxnID& txnid, const vector<CSCTypesList>
 inline void allocateValArray(void*& valArray, ColTupleList::size_type totalRow,
                              ColType colType, int colWidth)
 {
-    valArray = calloc(totalRow, colWidth);
+    switch (colType)
+    {
+        case WriteEngine::WR_VARBINARY : // treat same as char for now
+        case WriteEngine::WR_CHAR:
+        case WriteEngine::WR_BLOB:
+        case WriteEngine::WR_TEXT:
+            valArray = (char*) calloc(sizeof(char), totalRow * MAX_COLUMN_BOUNDARY);
+            break;
+        default:
+            valArray = calloc(totalRow, colWidth);
+            break;
+    }
```

Comment:
For WriteEngine::WR_TOKEN, is colWidth == sizeof(Token) ?

Reply:
Perhaps not. Hence the hard coded Constant "8"

Reply:
Sorry. I didn't parse your question. Now that I understand the question, the answer is: I can't definitely say. So I'll add token back into the case.